lib/spack/spack/spec.py

```diff
@@ -155,6 +155,10 @@
 # Special types of dependencies.
 alldeps = ('build', 'link', 'run')
 nolink = ('build', 'run')
+special_types = {
+    'alldeps': ('build', 'link', 'run'),
+    'nolink': ('build', 'run'),
+}
```

Comment:
@mathstuf Minor thing. What about : 
```python
special_types = {
    'alldeps': alldeps,
    'nolink': nolink,
}
```
if we are going to maintain both ways of referring to special deptypes ?